Automattic/wp-calypso
# Plugins: Move plugins data store to redux

This PR removes the plugin data flux store (the store with all the retrieved from .org API plugins info) and adds a redux subtree to store it.

The functionality is exactly the same, so for our users everything should look the same. The main difference is that both /plugins/ and /plugins/plugin-slug routes now store the plugin's info we fetch from WordPress.org plugins' API in the shared global state instead of using a dedicated flux store.

The PR only moves the .org data store, it doesn't touch any other plugin related stores or actions. The PR is quite large, but mainly because it moves code to different files (also, half of it are tests or needed boilerplate)

How to test
=========
1. Go to http://calypso.dev:3000/plugins
2. Make sure that your plugins load and you can see the icons
3. Click in any jetpack plugin
4. The plugin single view should load and you should be able to see all the plugin info there (icon, description, etc)
